src/libYARP_OS/src/Timer.cpp

```diff
@@ -0,0 +1,48 @@
+#include <yarp/os/Timer.h>
+#include <yarp/os/Time.h>
+using namespace yarp::os;
+
+Timer::Timer(const double rate, TimerCallback callback, Mutex* mutex, const bool inputOneshot) :
+    RateThread(rate),
+    m_callback(callback),
+    m_oneshot(inputOneshot),
+    m_mutex(mutex)
+{}
+
+
+bool Timer::threadInit()
+{
+    m_startStamp = yarp::os::Time::now();
```

Comment:
This timer will run on current clock, is there a way to have a timer that runs on the system clock even if a network clock is in use?